### DISCUSSION TITLE
Другие приложения для конфига

### DISCUSSION THREAD

**Discussion by re1ocate** [ImMALWARE/bash-warp-generator#57] [General]
Можно ли использовать другие приложения для конфига?
Пытаюсь найти те которые позволят выбирать приложения в которых будет использоваться туннель

**Comment by ImMALWARE** [comment]
AmneziaVPN, но у меня в нём не получилось

**Comment by re1ocate** [comment]
чекнул, там баг видимо, ждем фикса и будет юзабельно уже

**Comment by mts2200** [comment]
Можно. Стандартный [WireGuard](https://www.wireguard.com/install/), однако, [AmneziaWG](https://github.com/amnezia-vpn/amneziawg-windows-client/releases/) будет получше, что изначально и предложили. Есть ещё [WireSock](https://github.com/wiresock/WireSockUI/releases), но лично сам не пользовался, так что на свой страх и риск. Иногда для работы требуется запускать VPN от имени Администратора. Проксировать для конкретного приложения точно не знаю в каком можно.

UPD: Оказывается как-то он не совсем дружит с WireGuard, думал конфиги у них обратно совместимые.

**Comment by ImMALWARE** [comment]
> Стандартный [WireGuard](https://www.wireguard.com/install/)

Заблокирован у многих провайдеров.



> Оказывается как-то он не совсем дружит с WireGuard, думал конфиги у них обратно совместимые.

В конфиге AmneziaWG просто добавлены некоторые параметры, которые WireGuard считает неправильными.

**Comment by suzver** [comment]
WG Tunnel не подлючает с новой выходной нодой на т2, старый вариант (*engage.clo...*) работает. 
З.Ы. Убрал ИПВ6 адреса ноды и днс, заработало

**Comment by yaroslaff** [comment]
Скажите, а эти параметры:

~~~
S1 = 0
S2 = 0
Jc = 120
Jmin = 23
Jmax = 911
H1 = 1
H2 = 2
H3 = 3
H4 = 4
~~~
Они какие-то стандартные wireguard? Просто мой wireguard клиент и на андроиде (последний из google play) и на Raspbian 12 ругаются на них:
~~~
Line unrecognized: `S1=0'
~~~

(Да, с Amnezia все работает, я спрашиваю не чтобы заработало, а чтобы лучше понять). Или это параметры из AmneziaWG протокола?

И есть ли консольные клиенты, через которые это можно запустить на linux? Amnezia неудобно через SSH на роутере запускать :-) Может быть самому cloudflare-warp можно подсунуть этот конфиг? как?

**Comment by ImMALWARE** [comment]
> Скажите, а эти параметры:
> 
> ```
> S1 = 0
> S2 = 0
> Jc = 120
> Jmin = 23
> Jmax = 911
> H1 = 1
> H2 = 2
> H3 = 3
> H4 = 4
> ```
> 
> Они какие-то стандартные wireguard? Просто мой wireguard клиент и на андроиде (последний из google play) и на Raspbian 12 ругаются на них:
> 
> ```
> Line unrecognized: `S1=0'
> ```
> 
> (Да, с Amnezia все работает, я спрашиваю не чтобы заработало, а чтобы лучше понять). Или это параметры из AmneziaWG протокола?
> 
> И есть ли консольные клиенты, через которые это можно запустить на linux? Amnezia неудобно через SSH на роутере запускать :-) Может быть самому cloudflare-warp можно подсунуть этот конфиг? как?

В WireGuard это работать не будет, эти параметры специально для AmneziaWG

Насчет линукса, подписчик сделал гайд: https://telegra.ph/AmneziaWG-Dlya-Arch-Linux-11-15

**Comment by yaroslaff** [comment]
Спасибо, попробую!

**Comment by yaroslaff** [comment]
Нашел еще один (гораздо более легковесный) способ использования этих конфигов в Linux с консоли. Делюсь, может пригодится кому.

Есть проект: [wireproxy](https://github.com/pufferffish/wireproxy), он работает в userspace, устанавливает wireguard соединение и открывает socks прокси, через которыйможно пользоваться этим соединением (например, через [Proxy SwitchyOmega](https://chromewebstore.google.com/detail/proxy-switchyomega/padekgcemlokbadohgkifijomclgjgif)). 

У этого проекта есть форк [wireproxy-awg](https://github.com/artem-russkikh/wireproxy-awg) - то же самое, но с поддержкой AmneziaWG. И он легко подхватил этот конфиг, установил соединение!  

Программа легковесная, на golang, не требуется ничего патчить, устанавливать дополнительного, пересобирать каждый раз.

**Comment by suzver** [comment]
Для Андроид отличный клиент https://github.com/zaneschepke/wgtunnel
